R/dataframe-frame.R

```diff
@@ -127,19 +128,67 @@ dataframe__to_struct <- function(name = "") {
     wrap()
 }
 
+#' Convert an existing DataFrame to a LazyFrame
+#' @description Start a new lazy query from a DataFrame.
+#'
+#' @inherit as_polars_lf return
+#' @examples
+#' pl$DataFrame(a = 1:2, b = c(NA, "a"))$lazy()
 dataframe__lazy <- function() {
   self$`_df`$lazy() |>
     wrap()
 }
 
+#' Clone a DataFrame
+#'
+#' This is a cheap operation that does not copy data. This can be useful when
+#' dealing with attributes (see examples).
+#'
+#' @inherit as_polars_df return
+#' @examples
+#' df1 <- as_polars_df(iris)
+#'
+#' # Assigning doesn't change the memory address of the environment, but
+#' # cloning does.
```

Comment:
I think it is important to note that the environment object containing the DataFrame is not copied by assignment because it has *reference semantics*.
https://adv-r.hadley.nz/environments.html#explicit-envs

Reply:
Does "Assigning doesn't change the memory address of the environment because it has reference semantics, but cloning does." work for you?

If not can you suggest a replacement?

Reply:
How about that?
> Assigning does not copy the DataFrame (environment object). This is because environment objects have reference semantics.
> Calling `$clone()` creates a new environment.

Reply:
Sounds good, I think it fits better in the description than in the examples though. I'll move this